### ISSUE TITLE
PRB: build broken/blocked for arm64 by PR #7826

### ISSUE THREAD

**Issue by jrdodds** [dotnet/msbuild#7834]
### Issue Description
Building or just restoring fails with an error.
```
dotnet-install: The resource at legacy link 'https://dotnetbuilds.azureedge.net/public/Runtime/3.1.0/dotnet-osx-arm64.3.1.0.tar.gz' is not available.
dotnet_install: Error: Could not find `.NET Core Runtime` with version = 3.1.0
dotnet_install: Error: Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support
```

### Steps to Reproduce

1. Clone the repo to an Apple Silicon machine. (Commit 659a296 is where the issue is introduced.)
2. Execute `./build.sh --restore`

### Expected Behavior
Restore should complete without errors. 

### Actual Behavior
Restore fails -- which also blocks builds.

### Analysis
There is no arm64 version of runtime 3.10. PR #7826 changed global.json and introduced a dependency on runtime 3.10.

### Versions & Configurations
macOS Monterey 12.4; Apple M1 Pro (arm64)

**Comment by jrdodds** [comment]
Resolved by commit [a1d9d69](https://github.com/dotnet/msbuild/commit/a1d9d6928dbc16ab59bdcefdf13e5770077a07da)

**Comment by jrdodds** [comment]
Resolved
